pkg/stanza/fileconsumer/reader.go

```diff
@@ -124,12 +124,25 @@ func (r *reader) finalizeHeader() {
 	r.HeaderFinalized = true
 }
 
+// Delete will close and delete the file
+func (r *reader) Delete() {
+	if r.file == nil {
+		return
+	}
+	f := r.file
```

Comment:
why this assignment here?

Reply:
We need to close the file before we can delete it, but closing the file will now remove the handle saved on the reader. So this line just gives us a temporary handle to the file until we can delete it.

Reply:
ah yep.. there it is, i just needed to read line 145 :D thanks for the quick answer